src/magicbox/InputManager.ts

```diff
@@ -26,7 +27,12 @@ export class InputManager {
   public onchangecursor: () => void;
   public ontabpress: () => void;
 
-  constructor(element: HTMLElement, private onchange: (text: string, wordCompletion: boolean) => void, private magicBox: MagicBoxInstance) {
+  constructor(
+    element: HTMLElement,
+    private root: HTMLElement,
```

Comment:
This could be a breaking change if people using the search-ui are directly using or extending the InputManager.
If it's the case, we need to add new constructor properties at the end,